lib/job-iteration/iteration.rb

```diff
@@ -75,10 +75,6 @@ def around_iterate(&blk)
         set_callback(:iterate, :around, &blk)
       end
 
-      def interruption_adapter
-        JobIteration.interruption_adapter || JobIteration::InterruptionAdapters.lookup(queue_adapter_name)
-      end
```

Comment:
Having this method as a class method allowed the job class to overwrite it, which feels like it would be useful in tests, if nothing else (since we plan to get rid of the global `JobIteration.interruption_adapter`).

By moving the logic into the private method marked as `@private`, we've made it so it can't be controlled without violating the privacy contract.

Is that something we should reconsider?